src/server/worker/pipeline/transform/registry.go

```diff
@@ -0,0 +1,951 @@
+package transform
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/limit"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/client/pkg/pbutil"
+	"github.com/pachyderm/pachyderm/src/client/pps"
+	pfsserver "github.com/pachyderm/pachyderm/src/server/pfs"
+	"github.com/pachyderm/pachyderm/src/server/pkg/backoff"
+	col "github.com/pachyderm/pachyderm/src/server/pkg/collection"
+	"github.com/pachyderm/pachyderm/src/server/pkg/errutil"
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+	"github.com/pachyderm/pachyderm/src/server/pkg/ppsutil"
+	pfssync "github.com/pachyderm/pachyderm/src/server/pkg/sync"
+	"github.com/pachyderm/pachyderm/src/server/pkg/uuid"
+	"github.com/pachyderm/pachyderm/src/server/pkg/work"
+	"github.com/pachyderm/pachyderm/src/server/worker/common"
+	"github.com/pachyderm/pachyderm/src/server/worker/datum"
+	"github.com/pachyderm/pachyderm/src/server/worker/driver"
+	"github.com/pachyderm/pachyderm/src/server/worker/logs"
+)
+
+const maxDatumsPerTask = 10000
+
+func jobHashtreesTag(jobID string) string {
+	return fmt.Sprintf("job-%s-hashtrees", jobID)
+}
+
+type pendingJob struct {
+	client        *client.APIClient
+	commitInfo    *pfs.CommitInfo
+	cancel        context.CancelFunc
+	logger        logs.TaggedLogger
+	index         int
+	orphan        bool
+	ji            *pps.JobInfo
```

Comment:
Small nit, why not just call this `jobInfo`? You have `commitInfo`. I think this could also apply to `dit` -> `datumIterator`.